sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_lease_async.py

```diff
@@ -46,22 +42,30 @@ class ShareLeaseClient(LeaseClientBase):
         A string representing the lease ID of an existing lease. This value does not
         need to be specified in order to acquire a new lease, or break one.
     """
-
-    def __enter__(self):
-        raise TypeError("Async lease must use 'async with'.")
-
-    def __exit__(self, *args):
-        self.release()
+    def __init__(  # pylint: disable=missing-client-constructor-parameter-credential,missing-client-constructor-parameter-kwargs
```

Comment:
nit: Can we get a space after the comma since all your other compound disables follow that convention? I think this comment also applies to the sync version of this file. It is possible that maybe this is the character limit though? If so, then ignore this comment and just leave as-is

Reply:
Nice catch! Even though this disable is already over the line limit, it appears that comments don't count towards the line limit, so I'll add in a space after the comma between compound disables. :D

Reply:
For both lease sync and async!